main/parse.c

```diff
@@ -2409,28 +2410,48 @@ static bool processLangDefineKind(const langType language,
 	p += 2;
 	if (p[0] == '\0')
 		error (FATAL, "no kind name specified in \"--%s\" option", option);
-	tmp_end = strchr (p, ',');
-	if (!tmp_end)
+	marker_end = strchr (p, ',');
+	if (!marker_end)
 		error (FATAL, "no kind description specified in \"--%s\" option", option);
 
-	tmp_start = p;
-	while (p != tmp_end)
+	name_start = p;
+	while (p != marker_end)
 	{
-		if (!isalnum (*p))
-			error (FATAL, "unacceptable char as part of kind name in \"--%s\" option", option);
+		if (p == name_start)
+		{
+			if (!isalpha(*p))
+			{
+				char *name_in_msg = eStrndup (name_start, marker_end - name_start);
+				error (FATAL,
+					   "a kind name doesn't start with an alphabetical character: "
+					   "'%s' in \"--%s\" option",
+					   name_in_msg, option);
+			}
+		}
+		else
+		{
+			if (!isalnum (*p))
+			{
+				char *name_in_msg = eStrndup (name_start, marker_end - name_start);
+				error (FATAL,
+					   "other than alphabetical or numerical char is used as part of kind name: "
```

Comment:
I suggest:

    "non-alphanumeric char is used as part of kind name: "

Reply:
(Reflected the suggestoin to the local source tree).